### PR TITLE
Disables medical cyborg pinpointer and remote crew monitor access

### PR DESCRIPTION
**Pull request by tf-4**
## About The Pull Request

Disables the pinpointer upgrade from being printed. This also disables the remote crew monitor access.

## How This Contributes To The Skyrat Roleplay Experience

After a discussion yesterday about medical cyborgs being MDs, paramedics, and better at both at once, this was brought up as a potential way to resolve some of that. As it stands, they can locate a dead or dying crewmember extremely easily (better than a regular paramedic) through their environmental (fire, pressure, etc) resistances, their lack of slowdown, and their all-access. They can then do field surgery to revive the crew member on the spot. This takes a job away from TWO people (MD & paramedic) in favour of the work of ONE person (borg).

This'll stop medical cyborgs from being straight-up better paramedics, as they won't be able to find crew easily by themselves - though they could easily still team up with a paramedic (or anyone with a crew pinpointer/monitor) to take advantage of their innate AA and environmental resistances.

They can, if they really need to do paramedic work but cannot find a paramedic/willing crewmember to take with them, drag a modular tablet or such around with lifeline installed, or install it onto themselves. 

## Changelog
:cl:
balance: Medical cyborgs no longer get a pinpointer upgrade, nor the remote crew monitor access that comes with it.
/:cl:

### CODE REVIEW

**Comment by desertmote** [comment]
Would have proposed this too. I think the best solution is a medi and para borg both existing, but who's going to sprite that.

**Comment by hugmeorelse** [comment]
@tf-4 I understand the reason for this, but some borg players feel like a lot of recent changes have been taking a lot of things away from their repertoire while others feel like it's a sort of attack on them. I would consider that angle as well.

**Comment by tf-4** [comment]
> @tf-4 I understand the reason for this, but some borg players feel like a lot of recent changes have been taking a lot of things away from their repertoire while others feel like it's a sort of attack on them. I would consider that angle as well.

While I appreciate that, it isn't an attack on anyone. If we didn't do nerfs to avoid upsetting the mains of that role, we'd never nerf anything.

That said, I'm open to suggestions for light compensation buffs in another area.

**Comment by hugmeorelse** [comment]
> > @tf-4 I understand the reason for this, but some borg players feel like a lot of recent changes have been taking a lot of things away from their repertoire while others feel like it's a sort of attack on them. I would consider that angle as well.
> 
> While I appreciate that, it isn't an attack on anyone. If we didn't do nerfs to avoid upsetting the mains of that role, we'd never nerf anything.
> 
> That said, I'm open to suggestions for light compensation buffs in another area.

I'm opening to test this, but they're concerned that medical will creep into becoming a "time vampire" like Baymed. 

Again, I understand why we want to do this, but I would also consider alternative angles too to addressing this if possible.

**Review by GoldenAlpharex — APPROVED** [review]
I agree with this change, and although I'd say to bring it upstream, I don't think it's really as much of a problem for them considering how rare getting that upgrade is, and when you have it, you only have it for maybe twenty or thirty minutes. Here, you have it for like, an hour, an hour and a half, sometimes two hours. It's obviously a lot of time where you're basically strictly better than a Paramedic, to the point where there's little space left for them (the Paramedics) to have any use for most of the round.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Although you need to comment it out from the research nodes too.

**Comment by Xymec** [comment]
I don't mean to sound rude, BUT, just remove borgs at this rate. The pinpointer is mainly the only reason why people play medical borg. But please, if you think borgs are 'too strong', remove them already.

**Comment by OolongCow** [comment]
> That said, I'm open to suggestions for light compensation buffs in another area.

Add fired lollipops being added to the mask slot of the target if possible, an empty pocket if the mask slot is unavailable, and then any container with free space if neither of those can occur. Medical borgs are now saved.

Alternatively, just raise the research requirements so science is less inclined to get it almost incidentally. As far as I'm aware, it's laughably easy to get the research for it if you even remotely care about borg upgrades in general.

**Comment by tf-4** [comment]
> I don't mean to sound rude, BUT, just remove borgs at this rate. The pinpointer is mainly the only reason why people play medical borg. But please, if you think borgs are 'too strong', remove them already.

If you're just going to be hyperbolic without substance, I'll respectfully ignore your critique.

**Comment by OolongCow** [comment]
> Add fired lollipops being added to the mask slot of the target if possible, an empty pocket if the mask slot is unavailable, and then any container with free space if neither of those can occur. Medical borgs are now saved.

As an aside, this is also an emagged medical borg buff. Firing poisoned lollipops into people's mouths would be a pretty hilarious form of antaggery.

**Comment by Zergspower** [comment]
I'm not sure removing the medical tracker is necessary, mediborgs are already kicked out of the OR's which is why they became primarily paramedics and they're godawful at chemistry. Mirroring the comment from hugmeorelse, the repeated feature removals from the borg players is going to risk that playerbase leaving. And what happens in lowpop when there aren't any Paramedics? Would it make more sense to order the borg to stay in medbay and not retrieve bodies?

**Comment by Furminium** [comment]
From my view as someone who is borg a lot, borgs excel at what their model is for. Medical units are very good at healing crew, as an example. Yet, they make terrible engineers. Borgs are highly specialized, unlike crew that can simply use their hands to pick up metal if they wish to erect a wall. Just removing a module will be like shooting a light bulb to turn it off. It obviously works for the intended purpose but will bring more problems than it solves. It will mess with the borg player base as well as with people, as poking grille in maint can become round removal with ease. It is a problem that borgs are better parameds, save for a few parts regarding medical equipment but I rarely saw them beat each other up about it. It more often is a pacing problem, where medical capacity =/= dieing people. If you intend to fix borgs making their according departmental staff redundand, make cooperation valuable but not forced, their sets should complement each other but be able to stand alone as well.

This, in my opinion, adds to a different matter, lack of borg info and lore. Basically, what is a borg supposed to do past the obvious lawset. When can they research, should they do a surgery on the ground, space, a pillow or only in designated areas?
An example would be interruptions during a OP. If you make surgeon make a gross mistake if you bump into them during surgery, people will stay further away from surgeons, it will be more realistic, as this is apparently desired, and it will keep borgs from doing hallway surgeries.

tl;dr, removing stuff is a bad idea, work on the borg lore/soft rules, make pushed away surgeons gore their patients.

P.S.: If you are really so inclined, do a test merge, see if too much miasma builds up or if it is managable, I may be wrong after all.

**Comment by chizzyyyy** [comment]
Please do not remove this since it'll make playing mediborg absolutely painful for having no info on patients, it's already a balancing act of the whole suit sensors

**Comment by tf-4** [comment]
> If you intend to fix borgs making their according departmental staff redundand, make cooperation valuable but not forced, their sets should complement each other but be able to stand alone as well.

That's what this PR does. A medborg and paramed can team up to combine the paramedic's tracking with the borg's AA.
The borg can also stand alone as a powerful medical doctor, and the paramedic as a... Well, paramedic.

**Comment by tf-4** [comment]
> And what happens in lowpop when there aren't any Paramedics? Would it make more sense to order the borg to stay in medbay and not retrieve bodies?

The borg is welcome to team up with anyone that has hands in order to track down injured people. It would encourage cooperation rather than the borg being a one 'man' army.

**Comment by tf-4** [comment]
> Please do not remove this since it'll make playing mediborg absolutely painful for having no info on patients, it's already a balancing act of the whole suit sensors

I'm not sure what you mean here. Do you mean they won't be able to find injured people as easily? Because that's the point of the PR. Otherwise they can look at suit sensors just as easily as any medical doctor, on the consoles in medbay.

**Comment by GoldenAlpharex** [comment]
> Please do not remove this since it'll make playing mediborg absolutely painful for having no info on patients, it's already a balancing act of the whole suit sensors

You still have as much information on patient, the only thing you don't have is:
- Access to the Crew Monitor everywhere (you can still open the console in medbay, hell, I don't even know if you have a range requirement for it, if you do it's much bigger though).
- A pinpointer to locate bodies easily.

I really don't think this is the end of the world, especially considering cyborgs don't get it roundstart and are still very much able to play the role, so I don't see it as vital to their function.

And, well, no matter what you'll say, you can't exactly compare Medical Cyborgs to other modules aside from perhaps the Janiborg module, in that having one (in the janiborg's case, just having one suffices) with a certain module literally makes an entire job irrelevant unless there's *major* conflict happening on the station, which is like, once or twice a round usually.

And yes, I know what I'm on about, I'm a Paramedic main. I don't have jack shit to do when there's Mediborgs around, they literally fetch all the corpses before I even have the time to get out of Med. There's just no point, they're better at EVA, they get there faster, they have more access, you just can't compete. At least for Janitors, there's no magical "dirtiness pinpointer".

**Comment by Zergspower** [comment]
May I suggest allowing the proximity sensor to at least be added? The suggestion I saw about about dragging a tablet (that borgs cant interact with if I'm not mistake) isn't really feasible since they cant really buy their own. Or let them get it as software for their Robotact, this basically kills their ability to find bodies that are for example stuffed in a locker or God forbid the damn ice wastes where all the paramedics die on.  The nice thing about the health monitor is that it allowed borgs a chance to get out of medbay as well, vibe in the bar or whatever without shirking their duties.

**Comment by tf-4** [comment]
> May I suggest allowing the proximity sensor to at least be added? The suggestion I saw about about dragging a tablet (that borgs cant interact with if I'm not mistake) isn't really feasible since they cant really buy their own. Or let them get it as software for their Robotact, this basically kills their ability to find bodies that are for example stuffed in a locker or God forbid the damn ice wastes where all the paramedics die on. The nice thing about the health monitor is that it allowed borgs a chance to get out of medbay as well, vibe in the bar or whatever without shirking their duties.

They already can get it on their robotact

**Comment by Zergspower** [comment]
> > May I suggest allowing the proximity sensor to at least be added? The suggestion I saw about about dragging a tablet (that borgs cant interact with if I'm not mistake) isn't really feasible since they cant really buy their own. Or let them get it as software for their Robotact, this basically kills their ability to find bodies that are for example stuffed in a locker or God forbid the damn ice wastes where all the paramedics die on. The nice thing about the health monitor is that it allowed borgs a chance to get out of medbay as well, vibe in the bar or whatever without shirking their duties.
> 
> They already can get it on their robotact

IIRC only from getting a disk inserted into them, which no one knows or thinks to do honestly

**Comment by SpookyTheFox** [comment]
> > Please do not remove this since it'll make playing mediborg absolutely painful for having no info on patients, it's already a balancing act of the whole suit sensors
> 
> You still have as much information on patient, the only thing you don't have is:
> 
>     * Access to the Crew Monitor everywhere (you can still open the console in medbay, hell, I don't even know if you have a range requirement for it, if you do it's much bigger though).
> 
>     * A pinpointer to locate bodies easily.
> 
> 
> I really don't think this is the end of the world, especially considering cyborgs don't get it roundstart and are still very much able to play the role, so I don't see it as vital to their function.
> 
> And, well, no matter what you'll say, you can't exactly compare Medical Cyborgs to other modules aside from perhaps the Janiborg module, in that having one (in the janiborg's case, just having one suffices) with a certain module literally makes an entire job irrelevant unless there's _major_ conflict happening on the station, which is like, once or twice a round usually.
> 
> And yes, I know what I'm on about, I'm a Paramedic main. I don't have jack shit to do when there's Mediborgs around, they literally fetch all the corpses before I even have the time to get out of Med. There's just no point, they're better at EVA, they get there faster, they have more access, you just can't compete. At least for Janitors, there's no magical "dirtiness pinpointer".

I have a non trivial amount of hours as both paramedic and medborg. I have found that as a paramedic, I can easily compete with medical borgs, especially when I have jaws of life. The reason the borgs get there first, is because they don't call out the bodies, they just go get them. I can easily do the same thing as paramedic and leave no work for others. Infact, I do. And it's hard to find work, there are precious few high quality cases of death. That being, a situation where someone is dead or injured but on sensors and retrievable, and medbay as well as the station at large is in a state that doesn't preclude retrieving and reviving them.

I've done several experiments lately, much to the detriment of the station. If I don't get up and retrieve the bodies, weather as paramedic or borg, no one else will even if there are others in the role.

I feel like removing this only punishes the most attentive players who stay on task the whole round, and workarounds only reward the most experienced, while not solving the underlying problem of there just not being enough work to go around in the paramedic role. As is, paramedics have 40+ minutes of exclusive access to pinpointers and hand held crew monitors before borgs can be upgraded to have them.

Refocusing medborg on doctor work is not much of a consolation prize, their lack of hands greatly limits patient treatment strategies outside of surgery, and I can't speak for others but I frequently give that task up to organic doctors, and not always by choice. Due to recent changes to surgery, that too is degraded for borgs more so than doctors.

The underlying issue is the lack of patients for medical, not that borgs are abnormally good at a few select tasks. The other issue in play is how the role of paramedic is designed. On old base, it at least had access to the apothecary. They have very little to do, because their scope is so narrow now. They may be a premium antag role, but they have extraordinarily shallow gameplay in lieu of that. 

Same issue came up as one of several problems with the sec medic. Security is Medbay's primary customer and customer generator. While I was not permitted to remove that role outright, I did at least remove most of the late game medical items it spawned with as well as the tools to do surgery at round start, only to see many of my changes eventually rolled back via favorable map placement.

**Review by GoldenAlpharex — APPROVED** [review]
I'm down to give this a try.

**Comment by GoldenAlpharex** [comment]
We can revert this after a few days if it proves to not have the desired results.

**Comment by OolongCow** [comment]
> * Access to the Crew Monitor everywhere (you can still open the console in medbay, hell, I don't even know if you have a range requirement for it, if you do it's much bigger though).

You require LoS to it. It stops working the moment you break LoS.

Revisiting this PR, after playing several rounds of mediborg, my definite suggestion would be readding the pinpointer, but as the proximity version. They don't have an unfailing, perfect solution to finding crew that's better than a paramedic's, but they're still capable of doing so by using the console to find an initial location, then sussing the injured crewmember out using a proximity pinpointer once they arrive there.

**Comment by Zergspower** [comment]
Feedback:

I've done about 20 more hours of Mediborg since the push and here's my thoughts on it.

The crew monitor aspect is more of an annoyance than anything but manageable so long as the person is dead and not moving, the LoS requirement for it is fine _assuming_ the station has a public one (one CMO removed them all except the one in their office one shift... they never gave a reason). All in All I never saw an issue with having it and not having it isn't a huge loss.

My fears relating to the tracker were sadly true, with no paramedics alive or on staff IceBox recovery becomes extremely painful.  For two things:

1. Dragging a tracker makes the tracker needing to be left behind once you get to the body, i'm not sure what the hell changed but dragging a body onto your borg frame now makes it say their hands are full so you have to drag them behind you. This alone needs to be looked at because it's causing general recovery problems.
2. Lifeline's window is too damn big and works across Z-levels, meaning you spend forever trying to get to the spot only to realize they're a floor above/below. Plus the window real-estate takes up a 4th of your screen on top of the name box being non-sorted and the scan taking a few seconds.

Lifeline being dependent on someone uploading you with the disk was a larger pain in the ass than I anticipated, roboticists lately are more focused on MODSuits and either ignore borgs (last night I waited ten minutes with three of them ignoring me) to just get a battery upgraded.  Finding a disk due to research not researching it is a secondary issue, mainly since borgs cannot purchase a disk from the vendor for the 110 credits and no one wanted to be bothered to walk all the way to science and buy a disk for me.

If  above happens in some golden light aligns itself to where I can get lifeline early, it caused the same problems the PR was trying to get rid of.  I still don't believe the removal was necessary after playing again, I still see the same lazy parameds not paying attention to the monitor only this time the bodies are staying dead far long due to aimless wandering trying to find them.

All that said, Lifeline made a new meta of getting a pinpointer far earlier than paramedics would be able to, assuming you had a willing crewmember to A. have a tablet with access and B. had a disk to shove it in your face.   Without that, Search and Rescue especially on Icebox became a far larger issue and one shift spending nearly 50 minutes looking for three bodies.
